x/programs/cmd/simulator/src/lib.rs

```diff
@@ -95,17 +96,48 @@ pub enum Key {
 
 // TODO:
 // add `Cow` types for borrowing
-#[serde_as]
-#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
-#[serde(rename_all = "lowercase", tag = "type", content = "value")]
+#[derive(Clone, Debug, PartialEq)]
 pub enum Param {
-    U64(#[serde_as(as = "DisplayFromStr")] u64),
+    U64(u64),
     String(String),
     Id(Id),
-    #[serde(untagged)]
     Key(Key),
 }
 
+impl Serialize for Param {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut val = serializer.serialize_struct("param", 2)?;
```

Comment:
Haha this is definitely a big *nit*, but I would do this:
```suggestion
        let mut serializer = serializer.serialize_struct("param", 2)?;
```

Reply:
😨 
(replaced)